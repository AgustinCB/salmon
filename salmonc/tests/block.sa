print "BEFORE";
{
  print "ON THE BLOCK";
  {
    print "ASDASD";
  }
  var a = nil;
  print a;
  print a=3;
  print a+2;
}
print "AFTER";
