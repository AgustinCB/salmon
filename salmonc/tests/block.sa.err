Constants: [SimpleValue(Function { ip: 37, arity: 0, uplifts: None }), SimpleValue(Integer(1)), SimpleValue(Integer(4)), SimpleValue(Integer(3)), SimpleValue(String(0)), SimpleValue(String(1)), SimpleValue(Nil), SimpleValue(Function { ip: 51, arity: 0, uplifts: None }), SimpleValue(String(7)), SimpleValue(Integer(2)), SimpleValue(String(19)), SimpleValue(String(25)), SimpleValue(String(30))]
Instructions: [Instruction { instruction_type: Constant(0), location: 1 }, Instruction { instruction_type: SetGlobal(0), location: 1 }, Instruction { instruction_type: Pop, location: 1 }, Instruction { instruction_type: Constant(7), location: 5 }, Instruction { instruction_type: SetGlobal(1), location: 5 }, Instruction { instruction_type: Pop, location: 5 }, Instruction { instruction_type: Constant(4), location: 6 }, Instruction { instruction_type: Constant(10), location: 6 }, Instruction { instruction_type: ToStr, location: 6 }, Instruction { instruction_type: StringConcat, location: 6 }, Instruction { instruction_type: Push, location: 6 }, Instruction { instruction_type: Strlen, location: 6 }, Instruction { instruction_type: Swap, location: 6 }, Instruction { instruction_type: Constant(1), location: 6 }, Instruction { instruction_type: Constant(3), location: 6 }, Instruction { instruction_type: Constant(2), location: 6 }, Instruction { instruction_type: Syscall, location: 6 }, Instruction { instruction_type: Pop, location: 6 }, Instruction { instruction_type: Constant(6), location: 6 }, Instruction { instruction_type: GetGlobal(1), location: 6 }, Instruction { instruction_type: Call, location: 6 }, Instruction { instruction_type: Swap, location: 6 }, Instruction { instruction_type: Pop, location: 6 }, Instruction { instruction_type: Pop, location: 6 }, Instruction { instruction_type: Constant(4), location: 7 }, Instruction { instruction_type: Constant(11), location: 7 }, Instruction { instruction_type: ToStr, location: 7 }, Instruction { instruction_type: StringConcat, location: 7 }, Instruction { instruction_type: Push, location: 7 }, Instruction { instruction_type: Strlen, location: 7 }, Instruction { instruction_type: Swap, location: 7 }, Instruction { instruction_type: Constant(1), location: 7 }, Instruction { instruction_type: Constant(3), location: 7 }, Instruction { instruction_type: Constant(2), location: 7 }, Instruction { instruction_type: Syscall, location: 7 }, Instruction { instruction_type: Pop, location: 7 }, Instruction { instruction_type: Return, location: 7 }, Instruction { instruction_type: Constant(4), location: 1 }, Instruction { instruction_type: Constant(5), location: 1 }, Instruction { instruction_type: ToStr, location: 1 }, Instruction { instruction_type: StringConcat, location: 1 }, Instruction { instruction_type: Push, location: 1 }, Instruction { instruction_type: Strlen, location: 1 }, Instruction { instruction_type: Swap, location: 1 }, Instruction { instruction_type: Constant(1), location: 1 }, Instruction { instruction_type: Constant(3), location: 1 }, Instruction { instruction_type: Constant(2), location: 1 }, Instruction { instruction_type: Syscall, location: 1 }, Instruction { instruction_type: Pop, location: 1 }, Instruction { instruction_type: Constant(6), location: 1 }, Instruction { instruction_type: Return, location: 1 }, Instruction { instruction_type: Constant(4), location: 2 }, Instruction { instruction_type: Constant(8), location: 2 }, Instruction { instruction_type: ToStr, location: 2 }, Instruction { instruction_type: StringConcat, location: 2 }, Instruction { instruction_type: Push, location: 2 }, Instruction { instruction_type: Strlen, location: 2 }, Instruction { instruction_type: Swap, location: 2 }, Instruction { instruction_type: Constant(1), location: 2 }, Instruction { instruction_type: Constant(3), location: 2 }, Instruction { instruction_type: Constant(2), location: 2 }, Instruction { instruction_type: Syscall, location: 2 }, Instruction { instruction_type: Pop, location: 2 }, Instruction { instruction_type: Constant(6), location: 2 }, Instruction { instruction_type: GetGlobal(0), location: 2 }, Instruction { instruction_type: Call, location: 2 }, Instruction { instruction_type: Swap, location: 2 }, Instruction { instruction_type: Pop, location: 2 }, Instruction { instruction_type: Pop, location: 2 }, Instruction { instruction_type: Constant(6), location: 3 }, Instruction { instruction_type: SetLocal(0), location: 3 }, Instruction { instruction_type: Pop, location: 3 }, Instruction { instruction_type: Constant(4), location: 4 }, Instruction { instruction_type: GetLocal(0), location: 4 }, Instruction { instruction_type: ToStr, location: 4 }, Instruction { instruction_type: StringConcat, location: 4 }, Instruction { instruction_type: Push, location: 4 }, Instruction { instruction_type: Strlen, location: 4 }, Instruction { instruction_type: Swap, location: 4 }, Instruction { instruction_type: Constant(1), location: 4 }, Instruction { instruction_type: Constant(3), location: 4 }, Instruction { instruction_type: Constant(2), location: 4 }, Instruction { instruction_type: Syscall, location: 4 }, Instruction { instruction_type: Pop, location: 4 }, Instruction { instruction_type: Constant(4), location: 5 }, Instruction { instruction_type: Constant(3), location: 5 }, Instruction { instruction_type: SetLocal(0), location: 5 }, Instruction { instruction_type: ToStr, location: 5 }, Instruction { instruction_type: StringConcat, location: 5 }, Instruction { instruction_type: Push, location: 5 }, Instruction { instruction_type: Strlen, location: 5 }, Instruction { instruction_type: Swap, location: 5 }, Instruction { instruction_type: Constant(1), location: 5 }, Instruction { instruction_type: Constant(3), location: 5 }, Instruction { instruction_type: Constant(2), location: 5 }, Instruction { instruction_type: Syscall, location: 5 }, Instruction { instruction_type: Pop, location: 5 }, Instruction { instruction_type: Constant(4), location: 5 }, Instruction { instruction_type: GetLocal(0), location: 5 }, Instruction { instruction_type: Constant(9), location: 5 }, Instruction { instruction_type: Plus, location: 5 }, Instruction { instruction_type: ToStr, location: 5 }, Instruction { instruction_type: StringConcat, location: 5 }, Instruction { instruction_type: Push, location: 5 }, Instruction { instruction_type: Strlen, location: 5 }, Instruction { instruction_type: Swap, location: 5 }, Instruction { instruction_type: Constant(1), location: 5 }, Instruction { instruction_type: Constant(3), location: 5 }, Instruction { instruction_type: Constant(2), location: 5 }, Instruction { instruction_type: Syscall, location: 5 }, Instruction { instruction_type: Pop, location: 5 }, Instruction { instruction_type: Constant(6), location: 5 }, Instruction { instruction_type: Return, location: 5 }]
Locations: [Location { address: 30, line: 9 }, Location { address: 30, line: 4 }, Location { address: 30, line: 2 }, Location { address: 30, line: 6 }, Location { address: 30, line: 7 }, Location { address: 30, line: 8 }, Location { address: 30, line: 0 }, Location { address: 30, line: 11 }]
