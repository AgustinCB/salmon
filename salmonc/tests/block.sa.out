BEFORE
ON THE BLOCK
ASDASD
nil
3
5
AFTER
